pin/views.py

```diff
@@ -1,4 +1,316 @@
+from django.core            import serializers
+from django.db.models       import Q
+from django.core.exceptions import ObjectDoesNotExist
+
+from pin.models         import (
+    Topic,
+    UserTopic,
+    Board,
+    Pin,
+    Topic,
+    PinUser,
+    PinTopic,
+    Comment
+)
+from account.models         import User, Follow, Social
+from pinterrorist.settings  import S3URL, SECRET_KEY, ACCESS_KEY, SECRET_ACCESS_KEY
+
+def login_required(func):
+        def wrapper(self, request, *args, **kwargs):
+            
+            header_token            = request.headers['Authorization']
+            print(header_token)
+            if header_token         == None:
+                return JsonResponse({"message": "THE TOKEN IS EMPTY"}, status=402)
+            decoded_token           = jwt.decode(header_token, SECRET_KEY, algorithm='HS256')['id']
+            print(decoded_token)
+            user                    = User.objects.get(id=decoded_token)
+            user_id                 = user.id
+            kwargs['user']          = user
+            kwargs['user_id']       = user_id
+            
+            try:
+                if User.objects.filter(id=decoded_token).exists():
+                    return func(self, request, *args, **kwargs)
+                else:
+                    return JsonResponse({"message": "THE USER DOES NOT EXIST"})
+            except jwt.DecodeError:
+                return JsonResponse({"message": "WRONG_TOKEN!"}, status=403)
+            except KeyError:
+                return JsonResponse({"message": "KEY ERROR"}, status=405)
+        return wrapper
+
+class IntroTopicView(View):
+    
+    def get(self, request, *args, **kwargs):
+        topiclst        = list(Topic.objects.values())
+        return JsonResponse({"topics": topiclst})
+
+    @login_required
+    def post(self, request, *args, **kwargs):
+        body            = json.loads(request.body)
+        if body == None:
+            return JsonResponse({"message": "THERE IS NOTHING IN THE REQUEST BODY"}, status=405)
+        user            = kwargs['user']
+        user_id         = kwargs['user_id']
+        chosen_topics   = body['chosen_topics']
+        
+        topic_objs      = [Topic.objects.get(id=topic )for topic in chosen_topics]
+        for topic_obj in topic_objs:
+            user.topics.add(topic_obj)
+        return JsonResponse({'chosen_topics': chosen_topics}, status=200)
+
+class FileView(View):
+    s3_client = boto3.client(
+        's3',
+        aws_access_key_id=ACCESS_KEY,
+        aws_secret_access_key=SECRET_ACCESS_KEY
+    )
+
+    @login_required
+    def post(self, request, **kwargs):
+        
+        user        = kwargs['user']
+        user_id     = kwargs['user_id']
+        name        = request.POST.get('title')
+        paragraph1  = request.POST.get('text', None)
+        board       = request.POST.get('board', 1)
+        board       = Board.objects.get(id=board)
```

Comment:
board 로 입력된 값을 1이나 다른 수로 저장한 이후에 86라인에서 객체를 가져오게끔 되어있는데 default 가 1이므로 한줄로 써도 될것 같습니다. 다른줄에 쓰고 싶으면 아래와 같이 써도 될것같아요
board_id = request.POST.get('board', 1)
board       = Board.objects.get(id=board_id)
```suggestion
        board       = Board.objects.get(id=request.POST.get('board', 1))
```